vsutil.py

```diff
@@ -176,3 +195,75 @@ def is_image(filename: str, /) -> bool:
     Returns true if a filename refers to an image.
     """
     return types_map.get(path.splitext(filename)[-1], '').startswith('image/')
+
+
+def depth(clip: vs.VideoNode,
+          bitdepth: int,
+          /,
+          sample_type: Optional[Union[int, vs.SampleType]] = None,
+          *,
+          range: Optional[Union[int, Range]] = None,
+          range_in: Optional[Union[int, Range]] = None,
+          dither_type: Optional[Union[Dither, str]] = None) \
+        -> vs.VideoNode:
+    """
+    A bit depth converter only using core.resize and Format.replace.
+    By default, outputs FLOAT sample type for 32 bit and INTEGER for anything else.
+
+    :param bitdepth:    Desired bits_per_sample of output clip.
+    :param sample_type: Desired sample_type of output clip. Allows overriding default FLOAT/INTEGER behavior. Accepts
+                        vapoursynth.SampleType enums INTEGER and FLOAT or their values, [0, 1].
+    :param range:       Output pixel range (defaults to input clip's range). See `Range`.
+    :param range_in:    Input pixel range (defaults to input clip's range). See `Range`.
+    :param dither_type: Dithering algorithm. Allows overriding default dithering behavior. See `Dither`.
+        Defaults to Floyd-Steinberg error diffusion when downsampling, converting between ranges, or upsampling full
+        range input. Defaults to 'none', or round to nearest, otherwise.
+
+    :return: Converted clip with desired bit depth and sample type. ColorFamily will be same as input.
+    """
+    def readable_enums(enum, module: str = 'vsutil') -> list:
+        """
+        Returns a readable error message listing all possible values in `module.enum`.
+        """
+        return [module + "." + str(i) for i in enum] + [i.value for i in enum]
```

Comment:
For the record, an enum's `__repr__` would do it like this:
```py
['<Range.LIMITED: 0>', '<Range.FULL: 1>']
```

The only reason for not using repr directly here must be because one of the enums is from the vs module and it would thus be misleading where that could come from. Otoh, `vsutil` could reexport that enum (and list it in `__all__`).

Reply:
@FichteFoll getting the module name from `enum.__module__` doesn't work with imported modules.

```py
print(_readable_enums(vs.ColorFamily))

['importlib._bootstrap.ColorFamily.GRAY (1000000)', 'importlib._bootstrap.ColorFamily.RGB (2000000)', 'importlib._bootstrap.ColorFamily.YUV (3000000)', 'importlib._bootstrap.ColorFamily.YCOCG (4000000)', 'importlib._bootstrap.ColorFamily.COMPAT (9000000)']
```

Even calling it on `Range` gives `['__main__.Range.LIMITED (0)', '__main__.Range.FULL (1)']`

Reply:
Took FichteFoll's suggestion to extend the default `repr(enum.value)` behavior to prepend module name to the enum in 684a0415ba6a7e85f7606279ea55bc17009ba3a1.